### PR TITLE
[WIP] Login call reset password frontend

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest
The frontend needs to be changed so that we set the password after the registration has been done & the acount is activated.

### Issues

- relates #1130  

### Todo

- [ ] Register form withdraw password fields
- [ ] Refactor ResetPassword so that it works for /reset & /checkEmail
- [ ] Send them to thx/reset or thx/checkEmail

### CODE REVIEW

**Comment by ulfgebhardt** [comment]
@Elweyn Damn I haven't seen this in time - i made a basic implementation of the frontend to test the backend - so no text changes or any fancy hint texts like you did - we should try to combine both PRs as soon as possible in order to not make the work twice.
See: #1130

File: backend/src/graphql/resolver/UserResolver.ts [hunk 1]
```diff
@@ -440,20 +441,59 @@ export class UserResolver {
...
-  async sendResetPasswordEmail(
-    @Arg('email') email: string,
-  ): Promise<SendPasswordResetEmailResponse> {
-    const payload = {
-      email,
-      email_text: 7,
-      email_verification_code_type: 'resetPassword',
+  @Query(() => Boolean)
+  async sendResetPasswordEmail(@Arg('email') email: string): Promise<boolean> {
+    let emailAlreadySend = false
```
Comment (Mogge) [inline backend/src/graphql/resolver/UserResolver.ts:446]: I think we do not need this variable.

File: backend/src/graphql/resolver/UserResolver.ts [hunk 2]
```diff
@@ -440,20 +441,59 @@ export class UserResolver {
...
+
+    const loginUser = await LoginUser.findOneOrFail({ email })
+
+    let optInCode = await LoginEmailOptIn.findOne({
+      userId: loginUser.id,
+      emailOptInTypeId: EMAIL_OPT_IN_RESET_PASSWORD,
+    })
+    if (optInCode) {
+      emailAlreadySend = true
```
Comment (Mogge) [inline backend/src/graphql/resolver/UserResolver.ts:455]: Here the code from line 469 to 474

File: backend/src/graphql/resolver/UserResolver.ts [hunk 3]
```diff
@@ -440,20 +441,59 @@ export class UserResolver {
...
-    const response = await apiPost(CONFIG.LOGIN_API_URL + 'sendEmail', payload)
-    if (!response.success) {
-      throw new Error(response.data)
+
+    const link = CONFIG.EMAIL_LINK_SETPASSWORD.replace(
+      /\$1/g,
+      optInCode.verificationCode.toString(),
+    )
+
+    if (emailAlreadySend) {
```
Comment (Mogge) [inline backend/src/graphql/resolver/UserResolver.ts:469]: Move after the if clause

File: backend/src/graphql/resolver/UserResolver.ts [hunk 4]
```diff
@@ -440,20 +441,59 @@ export class UserResolver {
...
+      const timeElapsed = Date.now() - new Date(optInCode.updatedAt).getTime()
+      if (timeElapsed < 10 * 60 * 1000) {
+        throw new Error('email already sent less than 10 minutes before')
+      }
+    }
+
+    const emailSent = await sendEMail({
```
Comment (Mogge) [inline backend/src/graphql/resolver/UserResolver.ts:476]: TO DO: localize the email body and subject

File: frontend/src/locales/de.json [hunk 5]
```diff
@@ -145,6 +145,11 @@
         "text": "Jetzt kannst du ein neues Passwort speichern, mit dem du dich zukünftig in der Gradido-App anmelden kannst."
       },
       "send_now": "Jetzt senden",
+      "set": "Passwort festsetzen",
```
Comment (Mogge) [inline frontend/src/locales/de.json:148]: May be better `festlegen`

File: frontend/src/locales/en.json [hunk 6]
```diff
@@ -144,7 +144,12 @@
         "not-authenticated": "Unfortunately we could not authenticate you. Please contact the support.",
         "text": "Now you can save a new password to login to the Gradido-App in the future."
       },
-      "send_now": "Send now",
+      "send_now": "Jetzt senden",
+      "set": "Passwort festsetzen",
+      "set-password": {
+        "not-authenticated": "Leider konnten wir dich nicht authentifizieren. Bitte wende dich an den Support.",
```
Comment (Mogge) [inline frontend/src/locales/en.json:150]:
```suggestion
        "not-authenticated": "Unfortunately we could not authenticate you. Please contact the support.",
```

File: frontend/src/locales/en.json [hunk 7]
```diff
@@ -144,7 +144,12 @@
         "not-authenticated": "Unfortunately we could not authenticate you. Please contact the support.",
         "text": "Now you can save a new password to login to the Gradido-App in the future."
       },
-      "send_now": "Send now",
+      "send_now": "Jetzt senden",
```
Comment (Mogge) [inline frontend/src/locales/en.json:147]:
```suggestion
      "send_now": "Send now",
```

File: frontend/src/locales/en.json [hunk 8]
```diff
@@ -144,7 +144,12 @@
...
+      "set": "Passwort festsetzen",
```
Comment (Mogge) [inline frontend/src/locales/en.json:148]:
```suggestion
      "set": "Set password",
```

File: frontend/src/locales/en.json [hunk 9]
```diff
@@ -144,7 +144,12 @@
...
+      "set-password": {
+        "not-authenticated": "Leider konnten wir dich nicht authentifizieren. Bitte wende dich an den Support.",
+        "text": "Jetzt kannst du ein neues Passwort speichern, mit dem du dich zukünftig in der Gradido-App anmelden kannst."
```
Comment (Mogge) [inline frontend/src/locales/en.json:151]:
```suggestion
        "text": "Now you can save a new password to login to the Gradido-App in the future."
```

File: frontend/src/views/Pages/ResetPassword.vue [hunk 10]
```diff
@@ -111,9 +131,13 @@ export default {
       loader.hide()
       this.pending = false
     },
+    setDisplaySetup(from) {
+      this.displaySetup = textFields[this.$route.params.comingFrom]
+    },
   },
   mounted() {
```
Comment (Mogge) [inline frontend/src/views/Pages/ResetPassword.vue:138]:
```suggestion
  async mounted() {
```

File: frontend/src/views/Pages/ResetPassword.vue [hunk 11]
```diff
@@ -111,9 +131,13 @@ export default {
...
     this.authenticate()
```
Comment (Mogge) [inline frontend/src/views/Pages/ResetPassword.vue:139]:
```suggestion
    await this.authenticate()
```

**Review by Mogge — CHANGES_REQUESTED** [review]

**Comment by ulfgebhardt** [comment]
closing this since its now part of #1130
@Elweyn

**Pull request closed without merging**